lib/exabgp/reactor/api/command/text.py

```diff
@@ -32,22 +32,26 @@ def register (function):
 def _show_routes_callback(reactor, service, last, route_type, advertised, extensive):
 	def callback ():
 		families = None
+		lines_per_yield = environment.settings().api.chunk
 		if last in ('routes', 'extensive', 'static', 'flow', 'l2vpn'):
 			peers = reactor.peers.keys()
 		else:
 			peers = [n for n in reactor.peers.keys() if 'neighbor %s' % last in n]
+		line_count = 0
 		for key in peers:
 			peer = reactor.peers.get(key, None)
 			if not peer:
 				continue
 			if advertised:
 				families = peer._outgoing.proto.negotiated.families if peer._outgoing.proto else []
 			for change in list(peer.neighbor.rib.outgoing.sent_changes(families)):
```

Comment:
Some quick testing doesn't show much if any performance difference between the two. So I'll switch this to use slices instead.  Both gave me in the range of 9-12 seconds to dump 60k routes (20k at a time). For comparison before this patch it takes 30-40 seconds.